### ISSUE TITLE
select(multiple): aria-checked and aria-selected broken

### ISSUE THREAD

**Issue by npbenjohnson** [angular/material#10748]
**Actual Behavior**:
 -  `What is the issue? *`
`md-select` multiple values are always "checked" because of ngAria, and selected states don't clearly denote the control's intended usage.

 -  `What is the expected behavior?`
Only checked values should be checked. There is a workaround, adding `aria-checked` to `md-option` with a correct binding will prevent the ng-aria behavior. Ideally `md-select` would fix the ngAria behavior internally.

The distinction between selected and checked needs to be more clearly defined and expressed. According to specs, selection should be managed with `aria-activedescendant` unless the multiple select actually supports selecting (not checking) multiple values. `aria-selected` would be appropriate for a control where holding ctrl highlights (selects) multiple values, and hitting space checks them.

The issue is ambiguous because best practices for multiselects follow a pattern that is completely incompatible with `md-select` (`md-select` is actually more of a persistent-selectable tree-view than a multiselect based on examples I've seen). Expected behavior is a best-guess based on how I interpret selected and checked when both are present while using a screen reader.

Example of the confusion in actual vs expected read values by NVDA **for an unchecked item**:
Actual:
Foo item 1 of 3 checked not-selected
Expected:
Foo item 1 of 3 not checked

**Comment by benjiallen** [comment]
Related issue: #10969

**Comment by Splaktar** [comment]
Closed https://github.com/angular/material/issues/10969 as a duplicate of this one, but there are some helpful details in https://github.com/angular/material/issues/10969#issuecomment-354293608.

**Comment by Splaktar** [comment]
As mentioned in the OP, the `selected` attribute and `aria-selected` should be used with `aria-activedescendant` to manage virtual focus.

This is different than the value of the `md-select` which is the virtually focused item when the dropdown is closed or the list of checked options in a multiple select.

However, the current code is conflating both of these concepts into its definition of "selected" and ignoring that "checked" even exists.

**Comment by Splaktar** [comment]
Upon further testing and review, it does not appear that `aria-checked` has any role to play at all in `select` components. It appears that the `aria-selected` and `selected` attributes are what screen readers look for with `listbox` implementations. 

When the `select` has the `multiple` attribute, multiple options are allowed to have the `aria-selected` and `selected` attributes.

**Comment by Splaktar** [comment]
I've pushed a big update to PR https://github.com/angular/material/pull/11761 that gets rid of `aria-checked` and resolves a lot of other related issues. More testing, review, and feedback is needed. There are also some known issues listed in the PR under TODO.
